translate/pa3.1.md

```diff
@@ -0,0 +1,123 @@
+
+We have prepared the framework code of Nanos-lite for everyone, which can be obtained by executing the following command:
+
+    cd ics2023
+    bash init.sh nanos-lite
+    
+
+Nanos-lite contains all modules used in subsequent PAs, but the specific functions of most modules have not been implemented. Since the functions of the hardware (NEMU) are gradually added, Nanos-lite must also cooperate with this process. You will use some macros in `nanos-lite/include/common.h` which are used to control the functions of Nanos-lite during the experiment. As the experimental progress progresses, we will gradually explain all the modules, and Nanos-lite will do more and more work. Therefore, when reading the code of Nanos-lite, you only need to care about the modules related to the current progress, and do not get entangled in the code that is not related to the current progress.
+
+    nanos-lite
+    ├── include
+    │   ├── common.h
+    │   ├── debug.h
+    │   ├── fs.h
+    │   ├── memory.h
+    │   └── proc.h
+    ├── Makefile
+    ├── README.md
+    ├── resources
+    │   └── logo.txt    # Project-N logo text
+    └── src
+        ├── device.c    # Device abstraction
+        ├── fs.c        # File System
+        ├── irq.c       # Interrupt exception handling
+        ├── loader.c    # Loader
+        ├── main.c
+        ├── mm.c        # Storage management
+        ├── proc.c      # Process scheduling
+        ├── ramdisk.c   # ramdisk driver
+        ├── resources.S # ramdisk content and logo of Project-N
+        └── syscall.c   # System call handling
+    
+
+It should be reminded that Nanos-lite runs on AM, and AM’s APIs are available in Nanos-lite. Although the operating system is a special concept to us, from AM's perspective, it is just an ordinary C program that calls the AM API, no different from Super Mario. At the same time, you will once again realize the benefits of AM: the implementation of Nanos-lite can be architecture-independent, which means that no matter which ISA you have chosen before, you can easily run Nanos-lite, and even you can Just like developing klib, debug your Nanos-lite on `native`.
+
+In addition, although it will not cause obvious misunderstandings, after the introduction of Nanos-lite, we will still use the concept of "user process" instead of "user program" in some places. If you can't understand what a process is now, you only need to understand the process as a "running program". Still can’t feel the difference between the two? Let’s give a simple example. If you open Notepad three times, there will be three Notepad processes running on the computer, but there is only one Notepad program on the disk. Process is an important concept in the operating system. Detailed knowledge about processes will be introduced in the operating system class.
```

Comment:
```suggestion
In addition, although it will not cause obvious misunderstandings, after the introduction of Nanos-lite, we will still use the concept of "guest process" instead of "guest program" in some places. If you can't understand what a process is now, you only need to understand the process as a "running program". Still can’t feel the difference between the two? Let’s give a simple example. If you open Notepad three times, there will be three Notepad processes running on the computer, but there is only one Notepad program on the disk. Process is an important concept in the operating system. Detailed knowledge about processes will be introduced in the operating system class.
```